src/track-type-dependency.js

```diff
@@ -9,8 +10,21 @@ export default function trackTypeDependency(typeName) {
   types[typeName] = true;
 }
 
+export function trackFieldDependency(typeName, fieldName) {
+  if (!tracking) {
+    return;
+  }
+
+  if (!fields[typeName]) {
+    fields[typeName] = {};
+  }
+
+  fields[typeName][fieldName] = true;
```

Comment:
advantage to a hash is by not having to check for uniqueness. javascript doesn't have a builtin `uniq`, so we'd need to do a reduce on the built-out array.